.github/actions/gitops/gitops.sh

```
@@ -0,0 +1,59 @@
+# If you are using secrets to manage SSO metadata for Fleet SSO login or MDM SSO login, uncomment the below:
+
+# FLEET_SSO_METADATA=$( sed '2,$s/^/      /' <<<  "${FLEET_MDM_SSO_METADATA}")
+# FLEET_MDM_SSO_METADATA=$( sed '2,$s/^/        /' <<<  "${FLEET_MDM_SSO_METADATA}")
+
+# Copy/pasting raw SSO metadata into GitHub secrets will result in malformed yaml.
+# Adds spaces to all but the first line of metadata keeps the  multiline string in bounds.
+
+if compgen -G "$FLEET_GITOPS_DIR"/fleets/*.yml > /dev/null; then
+  # Validate that every fleet has a unique name.
+  # This is a limited check that assumes all fleet files contain the phrase: `name: <fleet_name>`
+  ! perl -nle 'print $1 if /^name:\s*(.+)$/' "$FLEET_GITOPS_DIR"/fleets/*.yml | sort | uniq -d | grep . -cq
```

Shellcheck (0.11.0)

[info] This ! is not on a condition and skips errexit. Use `&& exit 1` instead, or make sure $? is checked.

(SC2251)